app/lib/jsonvalidate.ts

```diff
@@ -0,0 +1,93 @@
+var fs = require('fs');
+var shell = require('shelljs');
+var check = require('validator');
+
+export class JsonValidate {
+	/*
+	 * Checks a json file for correct formatting against some validation function
+	 * @param jsonFilePath path to the json file being validated
+	 * @param jsonValidationFunction function that validates parsed json data against some criteria
+	 * @return the parsed json file
+	 * @throws InvalidDirectoryException if json file doesn't exist, InvalidJsonException on failed parse or *first* invalid field in json
+	*/
+	public validateJson(jsonFilePath: string, jsonValidationFunction: (path: string, jsonData: any) => void, jsonMissingErrorMessage?: string): any {
```

Comment:
I'm fine with the validation being another function, just wierd that the caller passes it.  It should internally call it.